File: packages/intune/android/src/main/java/io/capawesome/capacitorjs/plugins/intune/classes/options/DecryptFileOptions.java
package io.capawesome.capacitorjs.plugins.intune.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.intune.IntuneHelper;
import io.capawesome.capacitorjs.plugins.intune.classes.CustomExceptions;

public class DecryptFileOptions {

    @Nullable
    private final String destination;

    @NonNull
    private final String path;

    public DecryptFileOptions(@NonNull PluginCall call) throws Exception {
        this.destination = DecryptFileOptions.getDestinationFromCall(call);
        this.path = DecryptFileOptions.getPathFromCall(call);
    }

    @Nullable
    public String getDestination() {
        return destination;
    }

    @NonNull
    public String getPath() {
        return path;
    }

    @Nullable
    private static String getDestinationFromCall(@NonNull PluginCall call) {
        String destination = call.getString("destination");
        if (destination == null) {
            return null;
        }
        return IntuneHelper.getFilePath(destination);
    }

    @NonNull
    private static String getPathFromCall(@NonNull PluginCall call) throws Exception {
        String path = call.getString("path");
        if (path == null) {
            throw CustomExceptions.PATH_MISSING;
        }
        return IntuneHelper.getFilePath(path);
    }
}

File: packages/intune/ios/Plugin/Classes/Options/DecryptFileOptions.swift
import Foundation
import Capacitor

@objc public class DecryptFileOptions: NSObject {
    let destination: String?
    let path: String

    init(_ call: CAPPluginCall) throws {
        self.destination = DecryptFileOptions.getDestinationFromCall(call)
        self.path = try DecryptFileOptions.getPathFromCall(call)
    }

    private static func getDestinationFromCall(_ call: CAPPluginCall) -> String? {
        guard let destination = call.getString("destination") else {
            return nil
        }
        return IntuneHelper.getFilePath(destination)
    }

    private static func getPathFromCall(_ call: CAPPluginCall) throws -> String {
        guard let path = call.getString("path") else {
            throw CustomError.pathMissing
        }
        return IntuneHelper.getFilePath(path)
    }
}
